server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -546,14 +591,22 @@ public boolean equals(Object o) {
             return total == clusters.total && successful == clusters.successful && skipped == clusters.skipped;
         }
 
+        // TODO: should this be updated to include the new fields?
```

Comment:
I think we should wait until we have discussions with Tyler Perkins and the Kibana team about whether we are going to enhance the `_clusters` section in the REST response. It's possible these new fields will need to be written to the .async-search system index based on that decision. I think at that point we could look into extended the class for async search only.

Since this is an inner class of SearchResponse and Clusters is passed around a lot through the codebase (for async and sync search), that could be a fairly extensive refactoring to do, so my vote is to delay until we know what the long term plan is.

I'll add a note to the code about this.

Reply:
I see, I trust your judgement, I think I may have underestimated the effort needed. Let's mark the fields `transient` and ship it?!

Reply:
Good idea on marking them `transient`